server/memphis_handlers_tags.go

```diff
@@ -194,30 +278,103 @@ func (th TagsHandler) CreateTags(c *gin.Context) {
 	if !ok {
 		return
 	}
-	station_name, err := StationNameFromStr(body.EntityName)
-	if err != nil {
-		serv.Errorf("CreateTags error: " + err.Error())
-		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+	switch body.EntityType {
+	case "station":
+		station_name, err := StationNameFromStr(body.EntityName)
+		if err != nil {
+			serv.Errorf("CreateTags error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		exist, station, err := IsStationExist(station_name)
+		if err != nil {
+			serv.Errorf("CreateTags error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		if !exist {
+			serv.Warnf("Station does not exist")
+			c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "Could not create tags, station does not exist"})
+			return
+		}
+		err = AddTagsToEntity(body.Tags, body.EntityType, station.ID)
+		if err != nil {
+			serv.Errorf("Failed creating tag: %v", err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+	case "schema":
+		exist, schema, err := IsSchemaExist(body.EntityName)
+		if err != nil {
+			serv.Errorf("CreateTags error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		if !exist {
+			serv.Warnf("Schema does not exist")
+			c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "Could not create tags, schema does not exist"})
+			return
+		}
+		err = AddTagsToEntity(body.Tags, body.EntityType, schema.ID)
+		if err != nil {
+			serv.Errorf("Failed creating tag: %v", err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+	}
+
+	c.IndentedJSON(200, []string{})
+}
+
+func (th TagsHandler) CreateNewTag(c *gin.Context) {
+	var body models.CreateTag
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
 		return
 	}
-	exist, station, err := IsStationExist(station_name)
+	name := strings.ToLower(body.Name)
+	exist, _, err := IsTagExist(name)
 	if err != nil {
-		serv.Errorf("CreateTags error: " + err.Error())
 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
 		return
 	}
-	if !exist {
-		serv.Warnf("Station does not exist")
-		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "Could not create tags, station does not exist"})
+	if exist {
+		serv.Warnf("Tag with the same name already exists")
+		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "Tag with the same name already exists"})
 		return
 	}
-	err = AddTagsToEntity(body.Tags, body.EntityType, station.ID)
+	var newTag models.Tag
```

Comment:
why not use the logic that you created inside the createTag function it looks the same

Reply:
createTag creates a tag and adds an entity to it, createNewTag creates an empty new tag that is not connected to any entity yet - this is a ui requirement